app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -437,14 +446,20 @@ class HomeFragment : Fragment() {
         }
         homeViewModel.layoutManagerState =
             sessionControlView.view.layoutManager?.onSaveInstanceState()
-        homeViewModel.motionLayoutProgress = homeLayout?.progress ?: 0F
     }
 
     override fun onResume() {
         super.onResume()
         hideToolbar()
     }
 
+    override fun onPause() {
+        super.onPause()
+        val rect = Rect()
+        view!!.findViewById<AppBarLayout>(R.id.homeAppBar).getGlobalVisibleRect(rect)
```

Comment:
I'd prefer this to be in a separate function so the lifecycle method stays high level.

As per my comment above, maybe this should be in `onSaveInstanceState`